Fix bugs

SuhangXia opened the pull request:
## 动机 (Motivation)
Fix bugs
## 修改/新增的函数 (Modified/New Functions)
None
## 相关问题 (Related Issues)
## 变更类型 (Type of Change)
 - [x] Bugfix
 - [x] 功能 (Feature)
 - [ ] 文档 (Documentation)
 - [x] 测试 (Testing)
 - [ ] 重构 (Refactoring)
 - [ ] 其他 (Other) - 请说明 (Please specify)
## 变更详细信息 (Detailed Changes)
  1. Change the use of  ``inverse()`` to ``inverse_ext()`` in
  2. Add function content into ``Rotate()``.
  4. Add disconnect before connect in the function of ``Connect()``.
  5. Change functions of ``RecordInitialPos()`` and ``GoToInitialPos()`` from reference TCP to joint angles.
  6. Test all of the changes. Result: perfectly done.
## 影响范围 (Impact Scope)
```` 
Modules:
    MitkLancetHardwareDevice::DianaRobot.cpp
Plugins:
    org_mitk_lancet_hansrobot
````
## 测试结果 (Test Result)
Tested the code with Thor5 perfectly.

## 其他信息 (Additional Information)

SuhangXia committed:
Bug fix

Fix the funtion of CalculateForward();

SuhangXia committed:
Try to fix the of registration module

1. Change RecordInitialPos() from record the pose of TCP to joint angle.
2. Change  GoToInitialPos() from move by the initial pose of TCP to initial joint angle.

Those changes are needed a test!

SuhangXia committed:
Change functions of RecordInitialPos() and GoToInitialPos() in DianaRobot from reference TCP to joint angles.

Waiting for a test!

SuhangXia committed:
Add disconnect before connect in DianaRobot the function of Connect().

SuhangXia committed:
Change the use of inverse() to inverse_ext() in DianaRobot.

SuhangXia committed:
Bug fix about function of Rotate().

Czx1813260515 commented on Modules/LancetHardwareDevice/include/DianaRobot.h:
```diff
@@ -58,5 +58,6 @@ class MITKLANCETHARDWAREDEVICE_EXPORT DianaRobot : public AbstractRobot
 
 private:
 	double m_InitialPos[6] = { 0,0,0,0,0,0 };
```
这个不适合用于七轴的  机械臂 ，
#include <iostream>

// 模板抽象类，N表示关节的数量
template<size_t N>
class RobotArm {
public:
    // 构造函数，初始化关节角度为0
    RobotArm() {
        for (size_t i = 0; i < N; ++i) {
            joints[i] = 0.0;
        }
    }

    // 纯虚函数，要求子类实现
    virtual void moveJoints(const double newAngles[N]) = 0;

    // 获取当前关节角度
    const double* getJointAngles() const {
        return joints;
    }

    virtual ~RobotArm() {}

protected:
    double joints[N];  // 固定大小的关节角数组
};

// DianaRobot 实现
template<size_t N>
class DianaRobot : public RobotArm<N> {
public:
    // 使用基类的构造函数初始化关节
    DianaRobot() : RobotArm<N>() {}

    // 实现虚函数，移动关节
    void moveJoints(const double newAngles[N]) override {
        // 更新关节角度
        for (size_t i = 0; i < N; ++i) {
            this->joints[i] = newAngles[i];
        }

        // 显示更新后的角度
        std::cout << "DianaRobot moved joints to: ";
        for (size_t i = 0; i < N; ++i) {
            std::cout << this->joints[i] << " ";
        }
        std::cout << std::endl;
    }
};

Czx1813260515 commented on Modules/LancetHardwareDevice/include/DianaRobot.h:
```diff
@@ -58,5 +58,6 @@ class MITKLANCETHARDWAREDEVICE_EXPORT DianaRobot : public AbstractRobot
 
 private:
 	double m_InitialPos[6] = { 0,0,0,0,0,0 };
+	double m_jointsPos [6] = { 0,0,0,0,0,0 };
```
只用保留一个初始位的属性即可

Czx1813260515 commented on Modules/LancetHardwareDevice/include/LancetHansRobot.h:
```diff
@@ -74,12 +76,13 @@ class MITKLANCETHARDWAREDEVICE_EXPORT LancetHansRobot : public AbstractRobot
 	std::vector<double> CalculateInverse(Eigen::Vector3d aTranslation, Eigen::Vector3d aEulerAngle);
 
-	std::vector<double> CalculateForward(std::vector<double> aJointAngles);
+	//std::vector<double> CalculateForward(std::vector<double> aJointAngles);
```
相同的函数名  删掉即可

Czx1813260515 commented on Modules/LancetHardwareDevice/include/LancetHansRobot.h:
```diff
@@ -74,12 +76,13 @@ class MITKLANCETHARDWAREDEVICE_EXPORT LancetHansRobot : public AbstractRobot
 
 private:
 	vtkSmartPointer<vtkMatrix4x4> m_InitialPos;
+	vector<double> m_InitialJointAngle;
```
如果打算在 虚函数类中设置  这边就无需设置

Czx1813260515 commented on Modules/LancetHardwareDevice/src/LancetHansRobot.cpp:
```diff
@@ -3,7 +3,11 @@
 	m_FlangeToTCP = vtkSmartPointer<vtkMatrix4x4>::New();
+	m_FlangeToTCP->Identity();
+	PrintDataHelper::CoutMatrix("m_FlangeToTCP_Identity_matrix", m_FlangeToTCP);
+	std::cout << "LancetHansRobot���캯������" << std::endl;
```
std::cout  输出改成英文的 structure

Czx1813260515 commented on Modules/LancetHardwareDevice/src/LancetHansRobot.cpp:
```diff
@@ -56,7 +60,8 @@ void LancetHansRobot::Rotate(double* aDirection, double aAngle)
 void LancetHansRobot::RecordInitialPos()
 {
-	m_InitialPos->DeepCopy(GetBaseToTCP());
+	int ret = HRIF_ReadActJointPos(0,0, m_InitialJointAngle[0], m_InitialJointAngle[0], m_InitialJointAngle[0], m_InitialJointAngle[0], m_InitialJointAngle[0], m_InitialJointAngle[0]);
```
为啥都是 m_InitialJointAngle[0]

SuhangXia committed:
Merge branch 'Czx1813260515:master' into master

SuhangXia committed:
Merge branch 'Czx1813260515:master' into master

SuhangXia committed:
1111

SuhangXia committed:
Merge branch 'bug-fix'

SuhangXia committed:
Bug fix waiting for test.

Use replace the use of array to vector for robot joints numbers, in order to fitness the dynamic joints numbers for different robot arms.

SuhangXia committed:
bug fix

SuhangXia committed:
Return change

SuhangXia committed:
Update RobotArmRegistrationTab.cpp
